Wallet locking/unlocking

Mingling94 opened the pull request:
Wallet locking/unlocking work

TODO after PR, prettify notification symbols involved

seveibar commented:
Great job!

DavidVorick commented:
don't merge before the build finishes. Sometimes dumb things surprise you.

It's bitten Luke more than once.

Mingling94 committed:
Removed some uneccesary package.json stuff

Also added markup for more wallet popups

Mingling94 committed:
Reorganized wallet code

This will probably need some heavy linting

Mingling94 committed:
Added more error details to the api call error

Mingling94 committed:
Wallet password handling finally works

Woo, now just to get transaction history and stuff working

Mingling94 committed:
Lock and unlock are now explicit user commands

Also upgraded to font awesome 4.4

Mingling94 committed:
Split code further, reorganized

Locking/Unlocking now works perfectly *crosses fingers*

DavidVorick commented:
can we grab the minified version of font-awsome?

Thanks for PR-ing this!

Mingling94 committed:
Bulleted out password and clear on attempt

Mingling94 commented:
Removed the unminified font-awesome from my last commit node.

Probably an improper use of `git commit --amend`

Mingling94 commented on app/js/pluginManager.js:
```diff
@@ -60,8 +60,6 @@ function PluginManager() {
 			plugin.on('did-finish-load', plugin.show);
 			current = plugin;
-		} else {
-			plugin.on('did-finish-load', plugin.hide);
```
Plugins now start out as hidden to avoid logic errors of `plugin.hide()` & `webview.stop()` logic executing for every plugin upon load.

Mingling94 commented on app/js/pluginFactory.js:
```diff
@@ -48,9 +48,10 @@ module.exports = {
 
 		// Have all plugins displaying UI's zoom by default
-		v.addEventListener('did-finish-load', function() {
+		v.addEventListener('dom-ready', function() {
```
dom-ready occurs earlier than did-finish-load. prevents flickering

Mingling94 commented on app/js/uiManager.js:
```diff
@@ -30,6 +30,11 @@ function UIManager() {
+		unlocked: 'unlock',
+		ongoing: 'hourglass-half',
+		started: 'hourglass-start',
+		finished: 'hourglass-end',
```
can't wait to style and use these notifications in a useful manner ^_^

DavidVorick commented:
The best thing to have done would have been a rebase that replaces the file you added in the original commit that you introduced it. I think it's relatively okay the way you did it though. The project is not yet stable enough that commit history is super important.

Mingling94 committed:
Linted & added plugin-level .jshintrc

Still need to get rid of some errors

Mingling94 committed:
Removed debug output and fixed hosting

Mingling94 committed:
Added debug output to investigate odd error

/wallet/history returns error of "marshalled unlock hash is the wrong
length"

Also concatenated lifecycle.js with update.js

DavidVorick commented on app/css/plugin-standard.css:
```diff
@@ -54,7 +54,7 @@ html, body {
 	display: inline-block;
 	border: 1px solid #ffffff;
-	border-radius: 4px;
+	border-ra *dius: 4px;
```
?

DavidVorick commented on app/js/daemonAPI.js:
```diff
@@ -38,8 +38,10 @@ function sendCall(url, type, args, callback) {
+			var errcode = textStatus + ' ' + jqXHR.status + ' ' + errorThrown;
+			var errmsg = jqXHR.responseText;
+			callback(errcode + ' ' + errmsg);
 		},
```
what is jqXHR?

DavidVorick replied:
probably needs a comment

Mingling94 replied:
jqueryXmlHttpRequest, yeah I'll add a comment

DavidVorick commented on app/js/daemonManager.js:
```diff
@@ -61,6 +61,9 @@ function DaemonManager() {
 	}
 
+	/**
+	 * Checks if there is an update available
```
I like the go strategy of including the function name in the function docstring

updatePrompt checks if there is an update available

sometimes it helps when grepping

Mingling94 replied:
It's not a jsdoc standard. Why would it help grepping to have the function name right above the function itself? Wouldn't a grep just register two lines instead of one?

DavidVorick replied:
I thought the same thing when I heard about the go standard. But you usually also get a bit of helpful text explaining the function, which is nice.

DavidVorick commented on app/plugins/Hosting/js/hosting.js:
```diff
@@ -91,7 +91,7 @@ function update() {
 
 // Called upon showing
-function show() {
+function start() {
```
not show/hide anymore?

Mingling94 replied:
I thought it was more semantic while differentiating from the UI-level `plugin.show()` that also adds a transition fade and stuff

DavidVorick commented on app/plugins/Hosting/js/hosting.js:
```diff
@@ -100,7 +100,7 @@ function show() {
 
 // Called upon transitioning away from this view
-function hide() {
+function stop() {
```
same

DavidVorick commented:
ok, done with comments

Mingling94 committed:
Properly display wallet addresses and transactions

I think the changes should speak for themselves but essentially the
two tables in the Wallet plugin should be populated based on a user's
transaction history.

What's not considered are addresses that the user hasn't had a
transaction be related with yet. Those are just forgotten and can't be
retrieved by API call

Mingling94 committed:
commented what jqXHR is

Mingling94 commented:
@DavidVorick I think I addressed all your comments. This PR should be ready until the API changes (I know there's disagreement about uppercase vs. lowercase values to be sent to and returned from the API

Mingling94 commented:
This pr should have a working wallet plugin now. I'll turn my attention towards a renting plugin
